frontend/packages/console-shared/src/hooks/perspectiveUtils.ts

```diff
@@ -1,7 +1,125 @@
+  PerspectiveType,
+  AccessReviewResourceAttributes,
+  checkAccess,
+} from '@console/dynamic-plugin-sdk';
+import { LoadedExtension, useExtensions } from '@console/plugin-sdk/src';
 import { USERSETTINGS_PREFIX } from '../constants';
 
 const PERSPECTIVE_VISITED_FEATURE_KEY = 'perspective.visited';
 
+enum PerspectiveVisibilityState {
+  Enabled = 'Enabled',
+  Disabled = 'Disabled',
+  AccessReview = 'AccessReview',
+}
+
+type PerspectiveAccessReview = {
+  required?: AccessReviewResourceAttributes[];
+  missing?: AccessReviewResourceAttributes[];
+};
+
+type PerspectiveVisibility = {
+  state: PerspectiveVisibilityState;
+  accessReview?: PerspectiveAccessReview;
+};
+
+type Perspective = {
+  id: string;
+  visibility: PerspectiveVisibility;
+};
+
 export const getPerspectiveVisitedKey = (perspective: PerspectiveType): string =>
   `${USERSETTINGS_PREFIX}.${PERSPECTIVE_VISITED_FEATURE_KEY}.${perspective}`;
+
+export const hasReviewAccess = async (accessReview: PerspectiveAccessReview) => {
+  let hasAccess: boolean = true;
+  const requiredAccessReview = accessReview?.required;
+  const missingAccessReview = accessReview?.missing;
+  const requiredPromises = requiredAccessReview?.map((resourceAttributes) =>
+    checkAccess(resourceAttributes),
+  );
+  const missingPromises = missingAccessReview?.map((resourceAttributes) =>
+    checkAccess(resourceAttributes),
+  );
+
+  requiredPromises?.length > 0 &&
+    (await Promise.all(requiredPromises)
+      .then((values) => {
+        // enable the prespective if all the access review checks are successful
+        hasAccess = hasAccess && values.every((val) => val.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = hasAccess && false;
+      }));
+
+  missingPromises?.length > 0 &&
+    (await Promise.all(missingPromises)
+      .then((values) => {
+        // enable perspective if atleast one access review check fails
+        hasAccess = hasAccess && !values.every((val) => val.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = hasAccess && true;
+      }));
+
+  return hasAccess;
+};
+
+export const useGetPerspectives = (): LoadedExtension<PerspectiveExtension>[] => {
+  const perspectiveExtensions = useExtensions<PerspectiveExtension>(isPerspective);
+  const [extensions, setExtensions] = React.useState<LoadedExtension<PerspectiveExtension>[]>([]);
+  const adminPerspective = perspectiveExtensions.find(
+    (perspectiveEx) => perspectiveEx.properties.id === 'admin',
+  );
+
+  const setResults = React.useCallback(
+    (newResult: LoadedExtension<PerspectiveExtension>) => {
+      setExtensions((oldResults: LoadedExtension<PerspectiveExtension>[]) => {
+        const arr = [...oldResults, newResult];
+        return _.uniqBy(arr, 'properties.id');
```

Comment:
Is this needed because the useEffect hook is called multiple times? You might want to check if the array contains this extension already and replace it in this case.